tests/integration/package/package_integ_base.py

```diff
@@ -129,3 +150,83 @@ def get_command_list(
         if resolve_s3:
             command_list = command_list + ["--resolve-s3"]
         return command_list
+
+
+class CdkPackageIntegBase(PackageIntegBase):
+    def setUp(self):
+        # Sythensizing a CDK app produces a Cloud Assembly. To simulate an actual working setup, we copy the CDK app
+        # from test_data to a scratch dir as if the scratch dir is the working directory.
+        # This is similar to the setup in BuildIntegBase
+        self.scratch_dir = str(Path(__file__).resolve().parent.joinpath(str(uuid.uuid4()).replace("-", "")[:10]))
+        shutil.rmtree(self.scratch_dir, ignore_errors=True)
+        os.mkdir(self.scratch_dir)
+
+        self.working_dir = tempfile.mkdtemp(dir=self.scratch_dir)
+
+    def tearDown(self):
+        self.working_dir and shutil.rmtree(self.working_dir, ignore_errors=True)
+        self.scratch_dir and shutil.rmtree(self.scratch_dir, ignore_errors=True)
+
+    def get_command_list(
+        self,
+        s3_bucket=None,
+        template=None,
+        template_file=None,
+        s3_prefix=None,
+        output_template_file=None,
+        use_json=False,
+        force_upload=False,
+        no_progressbar=False,
+        kms_key_id=None,
+        metadata=None,
+        image_repository=None,
+        image_repositories=None,
+        resolve_s3=False,
+        cdk_app=None,
+    ):
+        command_list = super().get_command_list(
+            s3_bucket=s3_bucket,
+            template=template,
+            template_file=template_file,
+            s3_prefix=s3_prefix,
+            output_template_file=output_template_file,
+            use_json=use_json,
+            force_upload=force_upload,
+            no_progressbar=no_progressbar,
+            kms_key_id=kms_key_id,
+            metadata=metadata,
+            image_repository=image_repository,
+            image_repositories=image_repositories,
+            resolve_s3=resolve_s3,
+        )
+
+        if cdk_app:
+            command_list = command_list + ["--cdk-app", str(cdk_app)]
+
+        return command_list
+
+
+class CdkPackageIntegPythonBase(CdkPackageIntegBase):
+    def setUp(self):
+        super().setUp()
+        self._create_virtual_env()
+
+    def tearDown(self):
+        super().tearDown()
+
+    def _create_virtual_env(self):
+        create_venv_command = ["python3", "-m", "venv", ".venv"]
```

Comment:
any python3 version is fine, or do we want to specify a python3 version?
Also, the integration test runs with venv https://github.com/aws/aws-sam-cli/blob/1c42df379b442bd8fa16fcc6090035bdc884aa56/appveyor.yml#L122
We should recover the venv back after this test case.

Reply:
any python3 version is fine. The venv is used only to install cdk dependencies that are required for synthesize. 
And since the command is run with `cwd` set as `self.working_dir` (which is a temp folder created in each `setUp()` call), the venv actually lives under `self.working_dir`